src/main/scala/com/cognite/sdk/scala/common/OAuth2.scala

```diff
@@ -157,7 +157,8 @@ object OAuth2 {
     def apply[F[_]](
         credentials: ClientCredentials,
         refreshSecondsBeforeExpiration: Long = 30,
-        maybeCacheToken: Option[TokenState] = None
+        maybeCacheToken: Option[F[Option[TokenState]]] =
+          None // can't use F.pure(None) here hence extra Option[] wrapper
     )(
         implicit F: Async[F],
         clock: Clock[F],
```

Comment:
Only solution I could think of would be something like this, but maybe it's not worth it. It also requires specify the `F[_]` type on the apply call

```
object Hello {
  def apply[F[_]: Async](
      maybeCacheToken: F[Option[Int]]
    ): F[Int] = {
    apply(_ => maybeCacheToken)
  }

  def apply[F[_]: Async](
      maybeCacheToken: (Applicative[F]) => F[Option[Int]] = (ap) => ap.pure(None)
    ): F[Int] = {
    val a: F[Option[Int]] = maybeCacheToken(implicitly[Applicative[F]])
    ???
  }
}

def foo = {
  val a = Hello[IO](IO.pure(None))(implicitly[Async[IO]])
  val b = Hello[IO]()(implicitly[Async[IO]])
}
```